lib/chef/target_io/train/file.rb

```diff
@@ -51,35 +54,37 @@ def open(file_name, mode = "r")
           if block_given?
             yield(io)
 
-            if (content != new_content) && !mode.start_with?("r")
-              __transport_connection.file(file_name).content = new_content # Need Train 2.5+
-            end
+            # Return name of new remote file to be used in later operations
+            file_name = write_file(file_name, new_content) if (content != new_content) && !mode.start_with?("r")
+            io.instance_eval <<~INJECT
+              def path
+                "#{file_name}"
```

Comment:
I"m surprised rubocop didn't complain about htis.

```suggestion
                file_name.to_s
```